symphony/bdk/core/service/message/message_service.py

```diff
@@ -0,0 +1,217 @@
+    file_type
+)
+from symphony.bdk.gen.pod_api.default_api import DefaultApi
+from symphony.bdk.gen.pod_api.message_api import MessageApi
+from symphony.bdk.gen.pod_api.message_suppression_api import MessageSuppressionApi
+from symphony.bdk.gen.pod_api.pod_api import PodApi
+from symphony.bdk.gen.pod_api.streams_api import StreamsApi
+from symphony.bdk.gen.pod_model.message_metadata_response import MessageMetadataResponse
+from symphony.bdk.gen.pod_model.message_receipt_detail_response import MessageReceiptDetailResponse
+from symphony.bdk.gen.pod_model.message_status import MessageStatus
+from symphony.bdk.gen.pod_model.message_suppression_response import MessageSuppressionResponse
+from symphony.bdk.gen.pod_model.stream_attachment_item import StreamAttachmentItem
+
+
+class MessageService:
+    def __init__(self, messages_api: MultiAttachmentsMessagesApi,
+                 message_api: MessageApi,
+                 message_suppression_api: MessageSuppressionApi,
+                 streams_api: StreamsApi,
+                 pod_api: PodApi,
+                 attachment_api: AttachmentsApi,
+                 default_api: DefaultApi,
+                 auth_session: AuthSession
+                 ):
+        self._messages_api = messages_api
+        self._message_api = message_api
+        self._message_suppression_api = message_suppression_api
+        self._streams_api = streams_api
+        self._pod_api = pod_api
+        self._attachment_api = attachment_api
+        self._default_api = default_api
+        self._auth_session = auth_session
+
+    async def list_messages(
+            self,
+            stream_id: str,
+            since: int = 0,
+            skip: int = 0,
+            limit: int = 50
+    ) -> [V4Message]:
+        params = {
+            'sid': stream_id,
+            'since': since,
+            'session_token': self._auth_session.session_token,
+            'key_manager_token': self._auth_session.key_manager_token,
+            'skip': skip,
+            'limit': limit
+        }
+        message_list = await self._messages_api.v4_stream_sid_message_get(**params)
+        return message_list.value
+
+    async def send_message(
+            self,
+            stream_id: str,
+            message: str,
+            data: str = '',
+            version: str = '',
+            attachment: [file_type] = None,
+            preview: [file_type] = None
+    ) -> V4Message:
+        attachment = attachment if attachment is None or type(attachment) is list else [attachment]
+        preview = preview if preview is None or type(preview) is list else [preview]
+        params = {
+            'sid': stream_id,
+            'session_token': self._auth_session.session_token,
+            'key_manager_token': self._auth_session.key_manager_token,
+            'message': message,
+            'data': data,
+            'version': version
+        }
+        if attachment is not None:
+            params['attachment'] = attachment
```

Comment:
I'd prefer to have everything related to `attachment` in a single place like: remove line 70 and have:
```
if attachment is not None:
    params['attachment'] = attachment if type(attachment) is list else [attachment]
```
Same for `preview`.